### PR TITLE
Do not send 'new actor' notifications on ticket creation

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !23862

Due to #8496, when a ticket is created, "new actor" notifications will be sent in addition to the "new ticket" notification that already contains this information.

So if there is 3 observers, following notifications will be received:
- user1 receives 'new observer' notification (observers = user1)
- user1 receives 'new observer' notification (observers = user1, user2)
- user2 receives 'new observer' notification (observers = user1, user2)
- user1 receives 'new observer' notification (observers = user1, user2, user3)
- user2 receives 'new observer' notification (observers = user1, user2, user3)
- user3 receives 'new observer' notification (observers = user1, user2, user3)
- user1 receives 'new ticket' notification
- user2 receives 'new ticket' notification
- user3 receives 'new ticket' notification

### CODE REVIEW

**Review by trasher — APPROVED** [review]

**Comment by cedric-anne** [comment]
I tried to reproduce same behaviour as in GLPI 9.5. However, this is not completely possible, as actors adding is not done at the same time. Indeed, in GLPI 9.5, on ticket update from UI actors were passed in `_itil_assign` input that was handled in `prepareInputForUpdate`. In GLPI 10.0, on ticket update from UI, actors are passed in `_actors` input that is handled in `post_updateItem`.

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -8405,17 +8406,43 @@ protected function updateActors()
...
                     $actor_obj::$items_id_2 => $actor['items_id'],   // ex 'users_id' => 1
                     'type'                  => $actortype,
                 ]);
+                if (
+                    $actortype === CommonITILActor::ASSIGN
+                    && (
+                        (!isset($this->input['status']) && in_array($this->fields['status'], $this->getNewStatusArray()))
+                        || (isset($this->input['status']) && in_array($this->input['status'], $this->getNewStatusArray()))
+                    )
+                    && in_array(self::ASSIGNED, array_keys($this->getAllStatusArray()))
```
Comment (orthagh) [inline src/CommonITILObject.php:8429]:
```suggestion
                    && in_array(self::ASSIGNED, array_keys($this->getAllowedStatusArray()))
```

File: src/CommonITILObject.php [hunk 2]
```diff
@@ -8405,17 +8406,43 @@ protected function updateActors()
...
+            }
             foreach ($added as $actor) {
                 $actor_obj = $this->getActorObjectForItem($actor['itemtype']);
-                $actor_obj->add($actor + [
+                $actor_obj->add($common_actor_input + $actor + [
                     $actor_obj::$items_id_1 => $this->fields['id'], // ex 'tickets_id' => 1
                     $actor_obj::$items_id_2 => $actor['items_id'],   // ex 'users_id' => 1
                     'type'                  => $actortype,
                 ]);
+                if (
```
Comment (orthagh) [inline src/CommonITILObject.php:8423]: a bunch of comments for the block would be welcomed.

**Review by orthagh — CHANGES_REQUESTED** [review]

File: src/CommonITILObject.php (hunk 2 above)
Comment (cedric-anne) [reply src/CommonITILObject.php:8423]: It was copied/pasted from `prepareInputForUpdate`.

**Comment by orthagh** [comment]
yes, it's ok now for the notification part i guess

**Review by trasher — APPROVED** [review]

**Pull request merged**